overlord/overlord.go

```diff
@@ -122,6 +130,17 @@ func (o *Overlord) Run() {
 	})
 }
 
+func (o *Overlord) ensureAfter(d time.Duration) {
+	time.AfterFunc(d, func() {
```

Comment:
I'm ok with changing the semantics but we need a clearer name, EnsureNoLater(), EnsureWithin(), EnsureAtLatestAfter()...   "After" otherwise is too different from what it usually mean in the stdlib, which is usually not "no later than after D"